probe.go

```diff
@@ -0,0 +1,325 @@
+package udnssdk
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"strings"
+	"time"
+)
+
+// SBTCService manages Probes
+type SBTCService struct {
+	client *Client
+}
+
+// ProbeAlertDataDTO wraps a probe alert response
+type ProbeAlertDataDTO struct {
+	PoolRecord      string    `json:"poolRecord"`
+	ProbeType       string    `json:"probeType"`
+	ProbeStatus     string    `json:"probeStatus"`
+	AlertDate       time.Time `json:"alertDate"`
+	FailoverOccured bool      `json:"failoverOccured"`
+	OwnerName       string    `json:"ownerName"`
+	Status          string    `json:"status"`
+}
+
+// ProbeAlertDataListDTO wraps the response for an index of probe alerts
+type ProbeAlertDataListDTO struct {
+	Alerts     []ProbeAlertDataDTO `json:"alerts"`
+	Queryinfo  QueryInfo           `json:"queryInfo"`
+	Resultinfo ResultInfo          `json:"resultInfo"`
+}
+
+// ProbeInfoDTO wraps a probe response
+type ProbeInfoDTO struct {
+	ID         string           `json:"id"`
+	PoolRecord string           `json:"poolRecord"`
+	ProbeType  string           `json:"type"`
+	Interval   string           `json:"interval"`
+	Agents     []string         `json:"agents"`
+	Threshold  int              `json:"threshold"`
+	Details    *ProbeDetailsDTO `json:"details"`
+}
+
+// ProbeDetailsLimitDTO wraps a probe
+type ProbeDetailsLimitDTO struct {
+	Warning  int `json:"warning"`
+	Critical int `json:"critical"`
+	Fail     int `json:"fail"`
+}
+
+// ProbeDetailsDTO wraps the details of a probe
+type ProbeDetailsDTO struct {
+	data   []byte
+	Detail interface{} `json:"detail,omitempty"`
+	typ    string
+}
+
+// Populate does magical things with json unmarshalling
+func (s *ProbeDetailsDTO) Populate(typ string) (err error) {
```

Comment:
I notice this is only being called explicitly in the test file. Is this a magical method to be called by `json.Unmarshal()`? Or is it being explicitly called in the terraform provider?